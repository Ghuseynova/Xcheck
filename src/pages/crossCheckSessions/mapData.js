const mapData = (databaseData) => {
  const { id, title, author, state, taskId, coefficient, startDate, endDate } = databaseData;
  return {
    key: id,
    id,
    title,
    author,
    state,
    taskId,
    coefficient,
    startDate,
    endDate,
  };
};

export default mapData;
