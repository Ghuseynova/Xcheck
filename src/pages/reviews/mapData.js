const mapData = (databaseData) => {
  const { id, author, state, taskTitle, requestor } = databaseData;
  const grade = databaseData.grade.total;

  return {
    key: id,
    title: id,
    author,
    state,
    taskTitle,
    grade,
    requestor,
  };
};

export default mapData;
