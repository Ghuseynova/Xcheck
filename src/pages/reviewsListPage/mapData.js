const mapData = (databaseData) => {
  const { id, author, state, task } = databaseData;
  return {
    key: id,
    title: id,
    author,
    state,
    task,
  };
};

export default mapData;
