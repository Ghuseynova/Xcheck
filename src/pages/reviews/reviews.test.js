import React from 'react';
import { shallow } from 'enzyme';
import Reviews from './reviews';
import reviewsData from './test-data';
import mapData from './mapData';

const initialData = reviewsData[0];
const result = mapData(initialData);

describe('Check mapData', () => {
	it('Check mapData', () => {
		expect(result.author).toEqual(initialData.author);
		expect(result.requestor).toEqual(initialData.requestor);
		expect(result.key).toEqual(initialData.id);
		expect(result.state).toEqual(initialData.state);
		expect(result.taskTitle).toEqual(initialData.taskTitle);
        expect(result.title).toEqual(initialData.id);
	});
});

describe('render ', () => {
	it('render Reviews', () => {
		const tree = shallow(<Reviews />);
		expect(tree).toMatchSnapshot();
	});
});