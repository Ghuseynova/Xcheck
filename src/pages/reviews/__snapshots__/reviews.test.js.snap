// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render  render Reviews 1`] = `ShallowWrapper {}`;
